dc/core/PixelStreamUpdater.cpp

```diff
@@ -75,21 +83,10 @@ QImage PixelStreamUpdater::getTileImage( const uint frameIndex,
     // to determine which frames can be deleted from the _frames map
     _requestedFrameIndex = std::max( _requestedFrameIndex, frameIndex );
 
-    // find the segments for the requested frame
-    deflect::FramePtr frame;
-    for( const auto& i : _frames )
-    {
-        if( i.first == frameIndex )
-        {
-            frame = i.second;
-            break;
-        }
-    }
-
-    if( !frame )
+    if( _frames.empty( ))
         throw std::runtime_error( "No frames yet" );
 
-    const auto& segment = frame->segments.at( _visibleSet[tileIndex] );
+    const auto& segment = _currentFrame()->segments.at( _visibleSet[tileIndex]);
```

Comment:
it's _lastFrame(), not current. frameIndex imo should be still considered, but I'm not sure what I'm missing here why only showing the last frame is correct.